### DISCUSSION TITLE
turntable-like rotation in viewer

### DISCUSSION THREAD

**Discussion by phillyard-aicradle** [pyvista/pyvista#2588] [Q&A] [answered]
Testing out different 3d point cloud viewers. trying to replicate what [pptk](https://heremaps.github.io/pptk/index.html) does in pyvista where I can click and drag and the viewpoint rotates in a turntable fashion. In pyvista, I have to use an extra step to rotate the view so that I'm not looking at a slanted view of the points. I'm new to this area, and am not coming up with the right words to describe what I'm trying to accomplish. Here are two screen recording of me viewing the same data set in pptk and pyvista. Is there something I need to set/enable on the plotter to get this kind of viewing ability in pyvista?

pptk
https://user-images.githubusercontent.com/64037014/167903658-cc2150ee-47f3-436b-9334-557a8b661fd1.mov

pyvista
https://user-images.githubusercontent.com/64037014/167903358-ab969028-6d19-45be-8402-bc8c47454157.mov

**Comment by banesullivan** [comment, marked as answer]
Have you tried enable the "terrain style" interaction? https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_terrain_style.html

```py
import pyvista as pv
from pyvista import examples

plotter = pv.Plotter(notebook=0)
plotter.add_mesh(examples.download_lidar())
plotter.show_axes()
plotter.enable_terrain_style()
plotter.show()  
```

https://user-images.githubusercontent.com/22067021/176063353-5f80807d-e173-4bfd-bc21-79b1f040a4de.mov

**Reply by phillyard-aicradle** [reply]
Hadn't tried that, but it's exactly what I was looking for. TY
